design/external-references.md

```diff
@@ -184,33 +198,50 @@ class ApplicationLoadBalancer {
 
 The following methods will be added to the `Stack` class:
 
 ```ts
-exportString(exportName: string, value: string, options?: ExportOptions): void
-```
+class Stack {
+  exportString(exportName: string, value: string, options?: ExportOptions): void;
+  importString(exportName: string, options?: ImportOptions): string;
+}
 
+interface ExportOptions {
+  description?: string
+}
 
-`exportString` creates an AWS CloudFormation Output with `Value` set to `value` and `Export` set to `exportName`. `options` includes `description`.
+interface ImportOptions {
+  type?: ResolveType // default is Synthesis
+  weak?: boolean; // default is "false"
+}
 
-```ts
-importString(exportName: string): string
+enum ResolveType {
+  Synthesis,
+  Deployment
+}
 ```
 
-`importString` returns a stringified token for `Fn::ImportValue(exportName)`. Technically this method can be static, but for the sake of discoverability and symmetry, it's defined next to `exportString`.
+The `exportString` method creates an AWS CloudFormation Output for this value assigned to this export name.
 
-```ts
-exportObject(exportName: string, obj: ISerializable): void
-```
+The `importString` method returns the value for this specific export name. When importing a value, users can specify the following `ImportOptions`:
 
-The `exportObject` method will invoke `obj.serialize` with a serialization context "bound" to this export name. This means the export name will be used as a prefix to all written keys. `writeObject` will be implemented with a nested serialization context that adds another component to the export name prefix.
+* If `type` is set to `Deployment`, the method will return an `Fn::ImportValue(exportName)` token. This means that CDK code cannot reason about the concrete value, which will only be resolved when the stack is deployed.
+* If `type` is set to `Synthesis` (default) the method will exercise an environmental context provider to look up the export value __during synthesis__. The concrete value will be propagated to the CDK app and can be reasoned about like any normal value.
+* The `weak` option is only relevant for synth-time resolution. If it is `true` (which is the default), the CDK will automatically embed a `Metadata` entry on the consuming resource with an `Fn::ImportValue`. This will force CloudFormation to take a strong reference on the export, even through the actual value is concretely resolved during synthesis. This ensures, for example, that the producing stack can't be deleted as long as there stacks consuming the exported values. This behavior (which is the default), can be disabled by settings `weak: false`, in which case the `Fn::ImportValue` will simply not be included.
```

Comment:
And so you're proposing to take an import on the whole exported value, right? Does that even work in the metadata section?

Reply:
It does! ❤️